controllers/glance_controller.go

```diff
@@ -162,6 +162,21 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			util.LogErrorForObject(helper, err, "Set after and calc patch/diff", instance)
 		}
 
+		if changed := helper.GetChanges()["metadata"]; changed {
+			// Patch wipes out all defaulted non-scalar fields from the spec (such as PasswordSelector)
+			// that were not explicitly included in the instance's CR YAML
+			// if err := r.Patch(ctx, instance, patch); err != nil && !k8s_errors.IsNotFound(err) {
```

Comment:
I vaguely remember similar issues. At that time the problem was that the previous r.Update call updated our local instance object with the result of the Update. At that time I could circumvent the issue by passing a DeepCopy of the instance there. We have  r.Update calls before this place at L132. So I'm wondering if that is related.